.github/scripts/check_formalities.sh

```diff
@@ -48,19 +52,63 @@ for commit in $(git rev-list HEAD ^origin/"$BRANCH"); do
 		RET=1
 	fi
 
+	echo
+	info "Checking subject:"
 	subject="$(git show -s --format=%s "$commit")"
-	if echo "$subject" | grep -q -e '^[0-9A-Za-z,+/_-]\+: ' -e '^Revert '; then
-		success "Commit subject line seems OK ($subject)"
-	elif echo "$subject" | grep -iq '^Translated using Weblate.*'; then
-		warn "Weblate commit subject line exception: $subject"
-	elif echo "$subject" | grep -iq '^Added translation using Weblate.*'; then
-		warn "Weblate commit subject line exception: $subject"
+	echo "$subject"
+
+	# Check subject format
+	if echo "$subject" | grep -iq -e '^Translated using Weblate.*' -e '^Added translation using Weblate.*'; then
+		warn "Weblate commit subject line exception"
+	elif echo "$subject" | grep -q -e '^[0-9A-Za-z,+/_-]\+: [a-z]' -e '^Revert '; then
+		success "Commit subject line format seems OK"
+	elif echo "$subject" | grep -q -e '^[0-9A-Za-z,+/_-]\+: [A-Z]'; then
+		err "First word after prefix in subject should not be capitalized"
+		RET=1
+	elif echo "$subject" | grep -q -e '^[0-9A-Za-z,+/_-]\+: '; then
+		# Handles cases when there's a prefix but the check for capitalization fails (e.g. no word
+		# after prefix)
+		err "Commit subject line MUST start with '<package name>: ' and be followed by a lower-case word"
+		RET=1
 	else
-		err "Commit subject line MUST start with '<package name>: ' ($subject)"
+		err "Commit subject line MUST start with '<package name>: '"
+		RET=1
+	fi
+
+	if echo "$subject" | grep -q '\.$'; then
+		err "Commit subject line should not end with a period"
 		RET=1
 	fi
 
+	# Check subject length
+	if [ ${#subject} -gt "$MAX_SUBJECT_LEN" ]; then
+		warn "Commit subject line is longer than $MAX_SUBJECT_LEN characters (is ${#subject})"
+		split_err "$MAX_SUBJECT_LEN" "$subject"
+	else
+		success "Commit subject line is $MAX_SUBJECT_LEN characters or less"
+	fi
+
+	echo
+	info "Checking body:"
 	body="$(git show -s --format=%b "$commit")"
+	echo "$body"
+	echo
+
+	# Check body line lengths
+	body_line_too_long=0
+	line_num=0
+	while IFS= read -r line; do
+		line_num=$((line_num + 1))
+		if [ ${#line} -gt "$MAX_BODY_LINE_LEN" ]; then
+			warn "Commit body line $line_num is longer than $MAX_BODY_LINE_LEN characters (is ${#line}):"
+			split_err "$MAX_BODY_LINE_LEN" "$line"
```

Comment:
I am not sure, if we want to be such strict for commit description, because something I put there what it fixes, which error I've experienced and that usually contains more than 75 characters on one row. :-(

Reply:
This is just a warning and not an error, i.e the job will still pass. Same for subject.